Revert "ESfunctions: replace ElementTree with lxml.etree" and fix AIP fileuuid parsing

hakamine opened the pull request:
- Revert commit https://github.com/artefactual/archivematica/commit/7c5da722c761d42371ff7ae4b583071a9c2f6b6f to avoid changes in the way METS files are indexed
- Add AIP fileuuid tests 
- Add changes in `namespaces.py` that fixes AIP fileuuid parsing

Connected to archivematica/Issues#619

hakamine committed:
Revert "ESfunctions: replace ElementTree with lxml.etree"

This reverts commit 7c5da722c761d42371ff7ae4b583071a9c2f6b6f.

hakamine committed:
ESFunctions: add AIP fileuuid parse tests

jraddaoui commented on src/archivematicaCommon/lib/namespaces.py:
```diff
@@ -45,6 +47,7 @@ def xml_find_premis(elem, path):
     """``find`` with PREMIS 2 fallback."""
     matches = elem.find(path, namespaces=NSMAP)
     if matches is None:
+        ElementPath._cache.clear()
```
I'd appreciate a note explaining why is this needed. Maybe at the module level, as it's done in almost all the functions.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+
+
+# test moved out of the class
+# to be able to use @pytest.mark.parametrize
```
I'd remove this comment.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+    # Set up mocked up functions
+    # replace _try_to_index() with a mock function that returns the
+    # dict indexed_data, with the fileuuids that _index_aip_files()
+    # obtained from the METS
```
A docstring will probably look better in here.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+    mets_file_path = os.path.join(THIS_DIR, "fixtures", metsfile)
+    uuid = aipuuid
+    sipName = "{}-{}".format(aipname, uuid)
+    identifiers = []
```
This four variables are only used in the function call bellow so you could avoid their declaration.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+        identifiers=identifiers,
+    )
+
+    # check number of indexed files is correct
```
Uppercase first for consistency.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+    # Check fileuuids are correct
+    for file_uuid in fileuuid_dict:
+        indexed_fileuuid = indexed_data[file_uuid["filePath"]]
+        assert indexed_fileuuid == file_uuid["FILEUUID"]
```
No need to declare `ìndexed_fileuuid` in here, it could all go in one line.

jraddaoui commented on src/archivematicaCommon/tests/test_elasticsearch_functions.py:
```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+
+    # check number of indexed files is correct
+    assert indexed_files_count == filecount
+    assert dummy_try_to_index.call_count == filecount
```
Are this checks needed for this test? I think we should only check the file UUID parsing to reduce the METS files.

jraddaoui reviewed (changes requested):
Thanks for the quick turnaround @hakamine, this is looking really great!

I've added some minor comments bellow, mostly cosmetic. My biggest concerns are to reflect why the `Elementpath._cacche_clear()` calls are needed and to reduce the METS files added for the test. For the later, maybe we could reuse some existing METS files or simplify the ones included in here, just having one file on each one with the less data required to be a a valid METS file.

hakamine committed:
Namespaces: clear xml.etree cache on map change

namespaces.py provide PREMIS 3/2 fallback for xml.etree find functions,
altering the namespaces map when falling back to PREMIS 2. Cache must
be cleared for this to work.
(Ref. https://stackoverflow.com/a/24872696/1572895)

hakamine commented:
HI @jraddaoui  , thank you very much for the code review. I think I addressed all comments, hopefully this is good to merge now.

jraddaoui reviewed (approved):
Thanks a lot @hakamine!

Please, for the next time, try to add the feedback changes in extra commits to make it easier for the reviewer, we can always merge with `--squash` after ;)
